common/persistence/visibility/store/elasticsearch/visibility_store.go

```diff
@@ -658,6 +668,15 @@ func (s *visibilityStore) convertQuery(namespace namespace.Name, namespaceID nam
 
 func (s *visibilityStore) setDefaultFieldSort(fieldSorts []*elastic.FieldSort) []elastic.Sorter {
 	if len(fieldSorts) == 0 {
+		if strings.HasPrefix(s.index, v2IndexPrefix) {
+			// Sort by the index sorting order defined in the index template.
+			return []elastic.Sorter{
+				elastic.NewFieldSort(searchattribute.CloseTime).Desc(),
+				elastic.NewFieldSort(searchattribute.StartTime).Desc(),
+				elastic.NewFieldSort(searchattribute.RunID).Desc(),
```

Comment:
consider to define a static var for this so it can be reused in both build and query time.

Reply:
Do you really need to set this? Wouldn't it be sorted by these fields even if they are not specified?